mono/metadata/reflection.c

```diff
@@ -13691,13 +13691,14 @@ mono_reflection_lookup_signature (MonoImage *image, MonoMethod *method, guint32
  * LOCKING: Take the loader lock
  */
 gpointer
-mono_reflection_lookup_dynamic_token (MonoImage *image, guint32 token, gboolean valid_token, MonoClass **handle_class, MonoGenericContext *context)
+mono_reflection_lookup_dynamic_token (MonoImage *image, guint32 token, gboolean valid_token, MonoClass **handle_class, MonoGenericContext *context, MonoError *error)
 {
-	MonoError error;
 	MonoDynamicImage *assembly = (MonoDynamicImage*)image;
 	MonoObject *obj;
 	MonoClass *klass;
 
+	mono_error_init (error);
+	
 	obj = lookup_dyn_token (assembly, token);
 	if (!obj) {
```

Comment:
Not sure what you mean.  It should throw if obj is null and `valid_token` is false?